packages/transaction-controller/src/utils/batch.ts

```diff
@@ -673,61 +674,99 @@ function wipeTransactionBatchById(
+ */
+function newBatchMetadata(
+  transactionBatchMeta: Omit<TransactionBatchMeta, 'time' | 'status'>,
+): TransactionBatchMeta {
+  return {
+    ...transactionBatchMeta,
+    time: Date.now(),
+    status: TransactionStatus.unapproved,
+  };
+}
+
 /**
  * Prepares the approval data for a transaction batch.
  *
  * @param options - The options object containing necessary parameters.
  * @param options.batchId - The batch ID for the transaction batch.
- * @param options.chainId - The chain ID of the transactions.
- * @param options.from - The sender's address.
- * @param options.isSimulationEnabled - A function to check if simulation is enabled.
- * @param options.nestedTransactions - The array of nested transactions.
- * @param options.networkClientId - The network client ID.
- * @param options.origin - The origin of the transaction batch.
- * @param options.update - The update function to modify the transaction controller state.
+ * @param options.request - The request object including the user request and necessary callbacks.
  * @returns The prepared transaction batch metadata.
  */
 async function prepareApprovalData({
   batchId,
-  chainId,
-  from,
-  isSimulationEnabled,
-  nestedTransactions,
-  networkClientId,
-  origin,
-  update,
+  request,
 }: {
   batchId: Hex;
-  chainId: Hex;
-  from: Hex;
-  isSimulationEnabled: () => boolean;
-  nestedTransactions: TransactionBatchSingleRequest[];
-  networkClientId: string;
-  origin?: string;
-  update: UpdateStateCallback;
+  request: AddTransactionBatchRequest;
 }): Promise<TransactionBatchMeta> {
+  const {
+    messenger,
+    request: userRequest,
+    isSimulationEnabled,
+    getGasFeeEstimates,
+    update,
+    getEthQuery,
+    getChainId,
+  } = request;
+
+  const {
+    from,
+    origin,
+    networkClientId,
+    transactions: nestedTransactions,
+  } = userRequest;
+
   if (!isSimulationEnabled()) {
     throw new Error(
       'Cannot create transaction batch as simulation not supported',
     );
   }
+  log('Preparing approval data for batch');
+  const chainId = getChainId(networkClientId);
 
   const { gasLimit } = await simulateGasBatch({
     chainId,
     from,
     transactions: nestedTransactions,
   });
 
-  const txBatchMeta: TransactionBatchMeta = {
+  const txBatchMeta: TransactionBatchMeta = newBatchMetadata({
     chainId,
     from,
     gas: gasLimit,
     id: batchId,
     networkClientId,
     origin,
     transactions: nestedTransactions,
-  };
+  });
+
+  const defaultGasFeeFlow = new DefaultGasFeeFlow();
+  const gasFeeControllerData = await getGasFeeEstimates({
+    networkClientId,
+  });
+
+  const gasFeeResponse = await defaultGasFeeFlow.getGasFees({
+    ethQuery: getEthQuery(networkClientId),
+    gasFeeControllerData,
+    messenger,
+    transactionMeta: {
+      ...txBatchMeta,
+      txParams: {
+        from,
+        gas: gasLimit,
+      },
+    },
+  });
+
+  txBatchMeta.gasFeeEstimates = gasFeeResponse.estimates;
```

Comment:
The nature of gas fee estimates is they change per block, so should this ticket also update the `GasFeePoller` to update the batch metadata overtime, or is that a separate dedicated ticket?